src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.clear();
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User();
+        validUser.setLogin("validUser123");
+        validUser.setPassword("qwertyQ123");
+        validUser.setAge(20);
+
+        User registeredUser = registrationService.register(validUser);
+        assertNotNull(registeredUser, "User should be registered");
+        assertEquals(validUser.getLogin(), registeredUser.getLogin());
+        assertEquals(validUser.getPassword(), registeredUser.getPassword());
+        assertEquals(validUser.getAge(), registeredUser.getAge());
+        assertEquals(1, Storage.people.size(), "User should be added to the storage");
+    }
+
+    @Test
+    public void register_loginTooShort_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("12345");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageBelow18_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(17);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", thrown.getMessage());
```

Comment:
The test `register_ageBelow18_notOk` is checking for age below 18. To cover all edge cases, add tests for a user who is exactly 18 (which you have done in `register_ageExactly18_ok`) and a user who is one year below the minimum age (17 in this case).